include/netlink/route/nexthop.h

```diff
@@ -61,11 +61,23 @@ extern int		rtnl_route_nh_str2flags(const char *);
 /*
  * nexthop encapsulations
  */
+extern int		rtnl_route_nh_get_encap_type(struct rtnl_nexthop *nh);
+
 extern int		rtnl_route_nh_encap_mpls(struct rtnl_nexthop *nh,
 						 struct nl_addr *addr,
 						 uint8_t ttl);
 extern struct nl_addr *	rtnl_route_nh_get_encap_mpls_dst(struct rtnl_nexthop *);
 extern uint8_t		rtnl_route_nh_get_encap_mpls_ttl(struct rtnl_nexthop *);
+
+
+extern int		rtnl_route_nh_get_encap_seg6_mode(struct rtnl_nexthop * );
+extern int		rtnl_route_nh_get_encap_seg6_srh(struct rtnl_nexthop *, void **);
+
+extern int		rtnl_route_nh_get_encap_seg6_local_action(struct rtnl_nexthop *);
+extern int		rtnl_route_nh_has_encap_seg6_local_attr(struct rtnl_nexthop *, int);
```

Comment:
The argument "attr" is an enum value from kernel headers, like `SEG6_LOCAL_FLAVORS`. If the user wants to use this API, they would have to include the kernel header. And since `SEG6_LOCAL_FLAVORS` was only added a short time ago, they might need to use the numerical value. OK, that's doable, but ugly.

How is this function to be used?

Reply:
In my case:
```
if (rtnl_route_nh_has_encap_seg6_local_attr(nh, SEG6_LOCAL_TABLE)) {
    table_id = rtnl_route_nh_get_encap_seg6_local_table(nh);
} else if (rtnl_route_nh_has_encap_seg6_local_attr(nh, SEG6_LOCAL_VRFTABLE)) {
    table_id = rtnl_route_nh_get_encap_seg6_local_vrftable(nh);
}
```